### PR TITLE
fix: preserve contributors order (resolves #3565)

### LINKED ISSUE #3565: Contributor names in metadata always being listed in alphabetical order
### Description

The recent release of https://github.com/pressbooks/pressbooks/issues/2749 has the overlooked side effect of enforcing alphabetical order for listed contributors in a given field. The following fields are affected in Book Info: Authors, Editors, Translators, Reviewers, Illustrators, and Contributors. The Chapter Authors field in Chapter Metadata has the same issue. 

Any contributors included are always listed alphabetically by name, unlike before where the order of names could be changed as needed.

If a book currently has contributors listed in a specific order in one of the affected fields, saving Book Info or the chapter will reset the listed names to alphabetical order.

### Steps to Reproduce

1. Try to list book or chapter authors (or book editors, translators, etc.) in a certain (non-alphabetical) order.
2. Save changes.

**Expected behavior:** A custom order of selected contributors in a field can be saved.

**Actual behavior:** Saving changes resets the order of selected contributors in a field to alphabetical order.

It is also possible to reproduce the issue by going to an existing test book with contributor names listed in non-alphabetical order and saving changes. Here's a video of this occurring with book author names after pressing save in Book Info:

https://github.com/pressbooks/pressbooks/assets/44182629/a246fde9-265f-4ddb-a86c-d4b39233d7e4

### PR DESCRIPTION
**Pull request by greatislander**
This PR replaces the contributor fields use of `@pressbooks/multiselect` with the new `@pressbooks/reorderable-multiselect`. Fix for #3565

### CODE REVIEW

File: templates/metaboxes/fields/reorderable-multiselect.blade.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<div class="form-field">
+	<pressbooks-reorderable-multiselect data-messages="{{ json_encode([
+		'Add' => __('Add', 'pressbooks'),
+		'Remove' => __('Remove', 'pressbooks'),
+		'Move Up' => __('Move Up', 'pressbooks'),
+		'Move Down' => __('Move Down', 'pressbooks'),
+		'Available Options' => __('Available options', 'pressbooks'),
+		'Moved to position $1' => __('Déplacé en position $1'),
+		'Removed $1 from selection' => __('Supprimer $1 de la sélection'),
+		'Added $1 to selection' => __('Ajouté $1 à la sélection')
```
Comment (fdalcin) [inline templates/metaboxes/fields/reorderable-multiselect.blade.php]: Is there any reason why those are defaulted to French? 🤔

**Review by fdalcin — CHANGES_REQUESTED** [review]
Thank you @greatislander. LGTM and testing locally it seems to be working as expected in the book info as well as in the chapter page. Contributors order is preserved and the component behaves as expected.

I noticed that a couple of translations are defaulted to French. Is there any reason for that?

**Comment by greatislander** [comment]
@fdalcin can you send me a screenshot?

File: templates/metaboxes/fields/reorderable-multiselect.blade.php (hunk 1 above)
Comment (greatislander) [reply templates/metaboxes/fields/reorderable-multiselect.blade.php]: Oops, I was testing those strings. They should be English. I'll fix that.

**Review by fdalcin — APPROVED** [review]

**Pull request merged**